src/core/date-formatter.ts

```diff
@@ -0,0 +1,476 @@
+/**
+ * Date Formatter - Handlebars-based date formatting for Seasons & Stars
+ */
+
+import type { CalendarDate } from './calendar-date';
+import type { SeasonsStarsCalendar } from '../types/calendar';
+
+export class DateFormatter {
+  private calendar: SeasonsStarsCalendar;
+  private templateCache: Map<string, Function> = new Map();
+  private static helpersRegistered: boolean = false;
+
+  constructor(calendar: SeasonsStarsCalendar) {
+    this.calendar = calendar;
+    // Only register helpers once globally to avoid unnecessary re-registration
+    if (!DateFormatter.helpersRegistered) {
+      this.registerCustomHelpers();
+      DateFormatter.helpersRegistered = true;
+    }
```

Comment:
I'd rather this boolean by used inside the method, so the method can be called over and over and only performs the registrations once, instead of having to check before every call of the method.